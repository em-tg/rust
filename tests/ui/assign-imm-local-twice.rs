//@ check-pass
fn test() {
    let v: isize;
    //~^ HELP consider making this binding mutable
    //~| SUGGESTION mut
    v = 1; //~ NOTE first assignment
    println!("v={}", v);
    v = 2; //~ WARNING cannot assign twice to immutable variable
           //~| NOTE cannot assign twice to immutable
           //~| NOTE on by default
    println!("v={}", v);
}

fn main() {
}
