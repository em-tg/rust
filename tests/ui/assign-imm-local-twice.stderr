warning[E0384]: cannot assign twice to immutable variable `v`
  --> $DIR/assign-imm-local-twice.rs:8:5
   |
LL |     v = 1;
   |     ----- first assignment to `v`
LL |     println!("v={}", v);
LL |     v = 2;
   |     ^^^^^ cannot assign twice to immutable variable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider making this binding mutable
   |
LL |     let mut v: isize;
   |         +++

warning: 1 warning emitted

For more information about this error, try `rustc --explain E0384`.
